### PR TITLE
Andrew Added testing command and cases for testing localization

### PR DESCRIPTION
**Pull request by AlbertOpus2507**
Added test cases for deep copy and also easier testing in root directory

### CODE REVIEW

**Comment by lhmcgann** [comment]
Edit this PR to merge back onto release-v2.0 rather than main branch

File: EStimLibrary.sln [hunk 1]
```diff
@@ -7,6 +7,10 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{7E35BD38-108
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "EStimLibrary", "src\EStimLibrary\EStimLibrary.csproj", "{23710686-EDC8-41FC-AAC8-52B002F488D6}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{3193A4CC-4C0F-4A71-BF98-B7EF41A15C5D}"
```
Comment (lhmcgann) [inline EStimLibrary.sln]: Revert these changes to the EStimLibrary.sln. The tests/EstimLibrary.UnitTests/EstimLibrary.UnitTests.sln should already have the 2 projects included. EStimLibrary.sln is to open the library only

File: src/EStimLibrary/Core/SpatialModel/LocalizationData.cs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿namespace EStimLibrary.Core.SpatialModel;
-
+using System.Collections.Generic;
```
Comment (lhmcgann) [inline src/EStimLibrary/Core/SpatialModel/LocalizationData.cs]: Style thing: put using statements first, then 2 newlines, then namespace, then 2 new lines, then the class

File: src/EStimLibrary/Core/SpatialModel/LocalizationData.cs [hunk 3]
```diff
@@ -10,6 +10,9 @@
 public record LocalizationData(IEnumerable<int> AreasFullyContaining,
     IEnumerable<int> AreasPartiallyContaining)
 {
+    // Modify the properties using ToList() to create shallow copies during initialization
```
Comment (lhmcgann) [inline src/EStimLibrary/Core/SpatialModel/LocalizationData.cs]: What did doing this fix?

File: tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
+using EStimLibrary.Core.SpatialModel;
+using Xunit;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace EStimLibrary.UnitTests.Core.SpatialModel
```
Comment (lhmcgann) [inline tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs]: Style thing: 2 newlines between using and namespace declarations, then also put a ; at the end of the namespace declaration line so you can get rid of the outermost {} and unindent everything by one

File: tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    /// Test class for the LocalizationData record.
+    /// </summary>
+    public class LocalizationDataTests
+    {
+        /// <summary>
+        /// Test the basic construction of LocalizationData and 
+        /// ensure immutability (encapsulation).
+        /// </summary>
+        [Fact]
+        public void LocalizationData_Constructor()
```
Comment (lhmcgann) [inline tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs]: You can remove the class name prefix from each test method name. Just start with the name of the method you're testing (e.g., Constructor or Merge). See the "Our Coding Practices" notebook page > As You Go... section > item 4 for more test method naming convention description

File: tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            // Arrange
+            var areasFullyContaining = new List<int> { 1, 2, 3 };
```
Comment (lhmcgann) [inline tests/EStimLibrary.UnitTests/Core/SpatialModel/LocalizationTests.cs]: Make sure to test different values for each method; try to think of edge cases! You can give your test methods the [Theory] flag instead of the [Fact] flag so you can use [InlineData(...)] or [MemberData(...)] flags to pass in multiple test values without needing to copy the body of the test method over and over again

**Review by lhmcgann — CHANGES_REQUESTED** [review]

**Review by bacchuongdaungo — COMMENTED** [review]
looks ok to me

File: src/EStimLibrary/Core/SpatialModel/LocalizationData.cs (hunk 3 above)
Comment (AlbertOpus2507) [reply src/EStimLibrary/Core/SpatialModel/LocalizationData.cs]: By doing this the merge method already working with the deep copied version of the input rather than the actual input itself, just another layer of protection.
